pkg/util/proxyconfig/merge.go

```diff
@@ -20,24 +20,37 @@ import (
 // provided, a comma-separated string of cluster-wide noProxy settings
 // are returned.
 func MergeUserSystemNoProxy(proxy *configv1.Proxy, infra *configv1.Infrastructure, network *configv1.Network, cluster *corev1.ConfigMap) (string, error) {
-	apiServerURL, err := url.Parse(infra.Status.APIServerURL)
-	if err != nil {
-		return "", fmt.Errorf("failed to parse API server URL")
-	}
-
-	internalAPIServer, err := url.Parse(infra.Status.APIServerInternalURL)
-	if err != nil {
-		return "", fmt.Errorf("failed to parse API server internal URL")
-	}
-
 	set := sets.NewString(
 		"127.0.0.1",
 		"localhost",
-		network.Status.ServiceNetwork[0],
-		apiServerURL.Hostname(),
-		internalAPIServer.Hostname(),
 	)
 
+	if len(infra.Status.APIServerURL) > 0 {
```

Comment:
return early on error seems like more Go-idomatic rather than the if-else blocks.